sdk/tables/Azure.Data.Tables/tests/TableClientTests.cs

```diff
@@ -157,6 +157,7 @@ public void CreatedEnumPropertiesAreSerializedProperly()
             Assert.That(dictEntity["PartitionKey"], Is.EqualTo(entity.PartitionKey), "The entities should be equivalent");
             Assert.That(dictEntity["RowKey"], Is.EqualTo(entity.RowKey), "The entities should be equivalent");
             Assert.That(dictEntity["MyFoo"], Is.EqualTo(entity.MyFoo.ToString()), "The entities should be equivalent");
+            Assert.That(dictEntity.Keys, Is.EquivalentTo(new[] { "PartitionKey", "RowKey", "MyFoo" }), "Only PK, RK, and user properties should be sent");
```

Comment:
👍 
Let's also add the test for serializing a `TableEntity` with `ToOdataAnnotatedDictionary` to validate the same scenario when it is not a custom entity.

Reply:
Done (I think).